### PR TITLE
Coverage: try/catch/finally completion values. Fixes gh-1742

### PR DESCRIPTION
**Pull request by rwaldron**

### CODE REVIEW

**Comment by rwaldron** [comment]
@syg heads up for failures in V8—these might be unexpected.

**Comment by leobalter** [comment]
@rwaldron After battle testing this PR, I took the liberty to fix the eval tests and add some others involving return completions.

There is still a chance to add tests for try-catch-finally messing up returns using break or continue, lol.

```js
function fn() {
  do {
    try {
      return 'NOT ME';
    } catch {}
    finally {
      break;
    }
  } while (false);
  return 42;
}

fn(); // 42;
```

File: test/language/statements/try/completion-values-fn-finally-abrupt.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  } finally {
+    count.finally += 1;
+    throw new Test262Error('finally'); // If F.[[Type]] is normal, set F to C.
+  }
+  return 'wat';
+};
+
+assert.throws(Test262Error, fn, '1: try Abrupt, catch Abrupt, finally Abrupt; Completion: finally');
+assert.sameValue(count.catch, 1, '1: catch count');
+assert.sameValue(count.finally, 1, '1: fiinally count');
```
Comment (bakkot) [inline test/language/statements/try/completion-values-fn-finally-abrupt.js:39]:
```suggestion
assert.sameValue(count.finally, 1, '1: finally count');
```

**Comment by rwaldron** [comment]
@bakkot nice catch!

**Pull request merged**